contracts/SignatureValidatorManager.sol

```diff
@@ -0,0 +1,226 @@
+     *                  0x44 to 0x64 - 32 bytes length of `bytes` parameter
+     *                  0x64 to 0x68 - 4 bytes of Signature selector
+     *                  0x68 to 0x88 - 32 bytes domain separator
+     *                  0x88 to 0xa8 - 32 bytes struct hash
+     *                  0xa8 to end - contains offset, length of bytes, and actual bytes containing signatures
+     *             Else:
+     *                 bytes containing signature data
+     *                 default validation flow will be used which will depend on the account implementation
+     *
+     */
+    function handle(address account, address sender, uint256 /* value */, bytes calldata data) external override returns (bytes memory) {
+        // Skip first 4 bytes of data as it contains function selector
+        (bytes32 messageHash, bytes memory signatureData) = abi.decode(data[0x4:], (bytes32, bytes));
+
+        address signatureValidatorHooksAddress = signatureValidatorHooks[account];
+        bytes memory prevalidationData;
+        bytes memory returnData;
+
+        if (signatureValidatorHooksAddress != address(0)) {
+            checkPermittedModule(signatureValidatorHooksAddress, MODULE_TYPE_SIGNATURE_VALIDATOR_HOOKS);
+            prevalidationData = ISafeProtocolSignatureValidatorHooks(signatureValidatorHooksAddress).preValidationHook(
+                account,
+                sender,
+                data
+            );
+        }
+
+        if (bytes4(data[0x64:0x68]) == SIGNATURE_VALIDATOR_SELECTOR) {
+            returnData = abi.encode(validateWithSignatureValdiator(account, sender, messageHash, data[0x68:]));
+
+            if (signatureValidatorHooksAddress != address(0)) {
+                ISafeProtocolSignatureValidatorHooks(signatureValidatorHooksAddress).postValidationHook(account, prevalidationData);
+            }
+            return returnData;
+        }
+
+        returnData = defaultValidator(account, messageHash, signatureData);
+        if (signatureValidatorHooksAddress != address(0)) {
+            ISafeProtocolSignatureValidatorHooks(signatureValidatorHooksAddress).postValidationHook(account, prevalidationData);
+        }
+        return returnData;
+    }
+
+    /**
+     * @notice A view function for default signature validation flow.
+     * @param account Address of the account whose signature is to be validated. Account should support function `checkSignatures(bytes32, bytes, bytes)`
+     * @param hash bytes32 hash of the data that is signed
+     * @param signatures Arbitrary length bytes array containing the signatures
+     */
+    function defaultValidator(address account, bytes32 hash, bytes memory signatures) internal view returns (bytes memory) {
+        bytes memory messageData = abi.encodePacked(
+            bytes1(0x19),
+            bytes1(0x01),
+            IAccount(account).domainSeparator(),
+            ACCOUNT_MSG_TYPEHASH,
+            abi.encode(keccak256(abi.encode(hash)))
```

Comment:
Should we mix in the account address in the message and not rely it adding it to its `domainSeparator`?

Reply:
In that case, I think we should build the domain separator. My point is that, right now, there is nothing preventing the `Account` implementation from using a domain separator like:

```json
{
  "name": "just a name"
}
```

Which would not mix in the chain ID or verifying contract address. With that in mind, I would propose that either:
- The `SignatureValidatorManager` builds a domain separator specific for this signature
- We additionally mix in the address and chain into the signed data

The first one is nice, as it removes the additional `domainSeparator` message from the interface (which kind of makes sense - since `SignatureValidatorManager` is technically specifying the signing domain right?)

Reply:
> The SignatureValidatorManager builds a domain separator specific for this signature

this is reasonable but then validation would be specific to domain defined by the validator and not the account. My thinking is that default validation flow should be using account's domain separator.

@mmv08 @rmeissner your thoughts?

Reply:
To reiterate what we discussed synchronously:
- `SignatureValidatorManager` should compute the domain separator (for compatibility reasons, using the [same implementation as the current `Safe` contracts](https://github.com/safe-global/safe-contracts/blob/b344bd866b99f8f51fae20ea2b98680d5219028d/contracts/Safe.sol#L384-L394)
- The type-hash should be adjusted to use the same one as the [current `Safe` contracts](https://github.com/safe-global/safe-contracts/blob/b344bd866b99f8f51fae20ea2b98680d5219028d/contracts/handler/CompatibilityFallbackHandler.sol#L37)

Reply:
resolving this conversation as changes have been pushed as discussed